src/twisted/python/_setup.py

```diff
@@ -96,6 +96,7 @@
         'pyasn1',
         'cryptography >= 0.9.1',
         'appdirs >= 1.4.0',
+        'pynacl >= 1.0.1',
```

Comment:
I have issues with how pynacl operates myself, but I found it to be the lesser evil.  There are other libraries that support *25519, but some only support curve25519, some only support ed25519, and most of them haven't been updated in at least a year or two.

How I see things is that pynacl is optional to the extent that cryptography and pyasn1 are optional.  They're not needed unless you want to use conch, and since *25519 is the default (and strongly encouraged) algorithm of OpenSSH it should be included when conch is.